puzzles/models.py

```diff
@@ -14,11 +14,14 @@ class Puzzle(models.Model):
     PENDING = 'PENDING'
     SOLVED = 'SOLVED'
     STUCK = 'STUCK'
-    STATUS_CHOICES = [SOLVING, PENDING, SOLVED, STUCK]
+    EXTRACTION = 'EXTRACTION'
+    ALL_STATUSES = [SOLVING, PENDING, SOLVED, STUCK, EXTRACTION]
+    # Users should only be able to change status to one of these 2.
+    VISIBLE_STATUS_CHOICES = [SOLVING, STUCK, EXTRACTION]
```

Comment:
"one of these _3_"

I was debating whether we should differentiate STUCK vs. EXTRACTION but I don't feel strongly either way so fine with having separate options for them.

Reply:
Addressed.